src/material/schematics/ng-update/migrations/legacy-imports-error.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Rule, SchematicContext, Tree} from '@angular-devkit/schematics';
+import {NodePackageInstallTask} from '@angular-devkit/schematics/tasks';
+import * as ts from 'typescript';
+
+/**
+ * "Migration" that logs an error and prevents further migrations
+ * from running if the project is using legacy components.
+ * @param onSuccess Rule to run if there are no legacy imports.
+ */
+export function legacyImportsError(onSuccess: Rule): Rule {
```

Comment:
I didn't go through the `DevkitMigration` for this schematic, because it isn't a standard migration schematic and the operations in it are pretty basic so it wasn't worth it to update the common infrastructure.